xun/functions/store/redis.py

```diff
@@ -0,0 +1,143 @@
+from .store import Store
+from .store import StoreDriver
+from .store import NamespacedKey
+from collections.abc import KeysView
+import pickle
+import redis
+
+
+class Redis(Store):
+    def __init__(self, host=None, port=6379, db=0):
+        self.host = host
+        self.port = port
+        self.db = db
+
+    @property
+    def driver(self):
+        try:
+            return self._driver
+        except AttributeError:
+            self._driver = RedisDriver(self.host, self.port, self.db)
+            return self._driver
+
+    def __getstate__(self):
+        return super().__getstate__(), (self.host, self.port, self.db)
+
+    def __setstate__(self, state):
+        super().__setstate__(state[0])
+        self.host, self.port, self.db = state[1]
+
+
+class RedisDriver(StoreDriver):
+    def __init__(self, host, port, db):
+        self.redis = redis.Redis(host=host, port=port, db=db)
+
+    def __contains__(self, key):
+        k = key_to_redis_key(key)
+        return self.redis.exists(k)
+
+    def __delitem__(self, key):
+        if key not in self:
+            raise KeyError('{}'.format(key))
+
+        k = key_to_redis_key(key)
+        self.redis.delete(k)
+
+    def __getitem__(self, key):
+        if key not in self:
+            raise KeyError('{}'.format(key))\
+
+        k = key_to_redis_key(key)
+        v = self.redis.get(k)
+        return pickle.loads(v)
+
+    def __iter__(self):
+        return (redis_key_to_key(k) for k in self.redis.scan_iter())
+
+    def __len__(self):
+        return sum(1 for _ in self.__iter__())
+
+    def __setitem__(self, key, value):
+        k = key_to_redis_key(key)
+        v = pickle.dumps(value)
+        self.redis.set(k, v)
+
+    def scan_namespace_iter(self, namespace):
+        """ Scan Namespace iterator
+
+        Redis scan for a namespace prefix
+
+        Returns
+        -------
+        iterator
+            An iterator over namespace hits
+        """
+        fmt = 'xun:{namespace}:*'
+        namespace_hex = pickle.dumps(namespace).hex()
+        pattern = fmt.format(namespace=namespace_hex)
+        return self.redis.scan_iter(pattern)
```

Comment:
Note that in this method you do not encode anything and it still works.
Which returns me to the points
1) there should be one and only one method that does all the dumps -> hex -> encode thing
2) are we sure that all the steps here are needed